### DISCUSSION TITLE
how to invoke functions with different http methods

### DISCUSSION THREAD

**Discussion by armannaj** [supabase/supabase#11753] [Questions]
I realised that supabase invoke method only uses POST method. However, I have built an edge function that handles REST apis (with help of Oak http framework).

I wasn't able to figure out how I can invoke a function with a different http method (such as GET, PATCH, DELETE, etc).

Something like this is desirable:

```
const { data, error } = await supabase.functions.invoke("func1", {
       method: "GET"
    });
```

An example [here](https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/restful-tasks/index.ts) shows that an edge function can handle different HTTP method, but it was never shown how to invoke it via supabase client.

**Comment by GaryAustin1** [comment]
You could just use a javascript fetch and set the authorization header if needed.

I don't see anything here or in supabase-js with a request for adding this feature.  You might generate such a request.

**Comment by armannaj** [comment]
Thanks, I decided to contribute to it and made some changes and raised [this pull-request](https://github.com/supabase/functions-js/pull/50)
